Uncaught TypeError: Cannot read property 'substr' of null

[Enter steps to reproduce below:]

1. ...
2. ...

**Atom Version**: 1.7.0
**System**: Microsoft Windows 10 Education
**Thrown From**: [settings-view](https://github.com/atom/settings-view) package, v0.235.1


### Stack Trace

Uncaught TypeError: Cannot read property 'substr' of null

```
At C:\Users\Chertpong\AppData\Local\atom\app-1.7.0\resources\app.asar\node_modules\settings-view\lib\package-card.js:709

TypeError: Cannot read property 'substr' of null
    at C:\Users\Chertpong\AppData\Local\atom\app-1.7.0\resources\app.asar\node_modules\settings-view\lib\package-card.js:709:80
    at C:\Users\Chertpong\AppData\Local\atom\app-1.7.0\resources\app.asar\node_modules\settings-view\lib\package-manager.js:376:51
    at C:\Users\Chertpong\AppData\Local\atom\app-1.7.0\resources\app.asar\node_modules\settings-view\lib\package-manager.js:403:20
    at exit (C:\Users\Chertpong\AppData\Local\atom\app-1.7.0\resources\app.asar\node_modules\settings-view\lib\package-manager.js:73:16)
    at triggerExitCallback (C:\Users\Chertpong\AppData\Local\atom\app-1.7.0\resources\app.asar\src\buffered-process.js:215:47)
    at ChildProcess.<anonymous> (C:\Users\Chertpong\AppData\Local\atom\app-1.7.0\resources\app.asar\src\buffered-process.js:237:18)
    at emitTwo (events.js:87:13)
    at ChildProcess.emit (events.js:172:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:200:12)
```

### Commands

```
     -5:42.5.0 settings-view:open (atom-text-editor.editor.is-focused)
     -5:36.6.0 core:close (atom-text-editor.editor.mini.is-focused)
     -5:34.5.0 settings-view:open (atom-text-editor.editor.is-focused)
  2x -5:31 core:backspace (atom-text-editor.editor.mini.is-focused)
     -4:55 application:open-your-keymap (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
     -4:07.7.0 settings-view:check-for-package-updates (atom-workspace.workspace.scrollbars-visible-always.theme-one-dark-syntax.theme-one-dark-ui)
```

### Config

```json
{
  "core": {}
}
```

### Installed Packages

```coffee
# User
atom-ternjs, v0.13.2
highlight-selected, v0.11.2
linter, v1.11.4
linter-eslint, vundefined
minimap, v4.21.0
minimap-highlight-selected, v4.4.0
open-recent, v5.0.0

# Dev
No dev packages
```

Win7: apm update did not work for me.

C:\Users\rpolisetti\.atom\packages>apm update
Package Updates Available (4)
├── atom-typescript 8.8.1 -> 8.9.1
├── minimap 4.21.0 -> 4.23.2
├── quick-highlight 0.3.10 -> 0.4.0
└── vim-mode-plus 0.34.0 -> 0.38.0

Would you like to install these updates? (yes) yes

Installing atom-typescript@8.9.1 to C:\Users\rpolisetti\.atom\packages failed

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Users\\rpolisetti\\AppData\\Local\\atom\\app-1.7.3\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\rpolisetti\\AppData\\Local\\atom\\app-1.7.3\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\rpolisetti\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\rpolisetti\\.atom\\.apmrc" "install" "C:\\Users\\RPOLIS~1\\AppData\\Local\\Temp\\d-11643-10440-jo0jdc\\package.tgz" "--target=0.36.8" "--arch=ia32" "--msvs_version=2013"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! file C:\Users\rpolisetti\.atom\.apm\atom-space-pen-views\2.2.0\package\package.json
npm ERR! code EJSONPARSE

npm ERR! Failed to parse json
npm ERR! Unexpected token '\u0000' at 1:1
npm ERR!
npm ERR! ^
npm ERR! File: C:\Users\rpolisetti\.atom\.apm\atom-space-pen-views\2.2.0\package\package.json
npm ERR! Failed to parse package.json data.
npm ERR! package.json must be actual JSON, not just JavaScript.
npm ERR!
npm ERR! This is not a bug in npm.
npm ERR! Tell the package author to fix their package.json file. JSON.parse

C:\Users\rpolisetti\.atom\packages>

**apm update did not work for me on Windows 7**

C:\>apm update
Package Updates Available (6)
├── atom-typescript 8.8.1 -> 10.1.5
├── linter 1.11.4 -> 1.11.14
├── minimap 4.21.0 -> 4.24.7
├── minimap-cursorline 0.1.0 -> 0.2.0
├── quick-highlight 0.3.10 -> 0.5.0
└── vim-mode-plus 0.34.0 -> 0.49.0

Would you like to install these updates? (yes) yes

Installing atom-typescript@10.1.5 to C:\Users\rpolisetti\.atom\packages failed

npm ERR! Windows_NT 6.1.7601
npm ERR! argv "C:\\Users\\rpolisetti\\AppData\\Local\\atom\\app-1.8.0\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\rpolisetti\\AppData\\Local\\atom\\app-1.8.0\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\rpolisetti\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\rpolisetti\\.atom\\.apmrc" "install" "C:\\Users\\RPOLIS~1\\AppData\\Local\\Temp\\d-116625-12820-1ct122l\\package.tgz" "--target=0.36.8" "--arch=ia32" "--msvs_version=2013"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! file C:\Users\rpolisetti\.atom\.apm\atom-space-pen-views\2.2.0\package\package.json
npm ERR! code EJSONPARSE

npm ERR! Failed to parse json
npm ERR! Unexpected token '\u0000' at 1:1
npm ERR!
npm ERR! ^
npm ERR! File: C:\Users\rpolisetti\.atom\.apm\atom-space-pen-views\2.2.0\package\package.json
npm ERR! Failed to parse package.json data.
npm ERR! package.json must be actual JSON, not just JavaScript.
npm ERR!
npm ERR! This is not a bug in npm.
npm ERR! Tell the package author to fix their package.json file. JSON.parse

C:\>

--- a/lib/package-card.coffee
+++ b/lib/package-card.coffee
@@ -467,10 +467,20 @@ class PackageCard extends View
 
   update: ->
     return unless @newVersion or @newSha
-    @packageManager.update @installablePack ? @pack, @newVersion, (error) =>
+    pack = @installablePack ? @pack
+    version = if @newVersion then "v#{@newVersion}" else "##{@newSha.substr(0, 8)}"
+    @packageManager.update pack, @newVersion, (error) =>
       if error?
-        version = if @newVersion then "v#{newVersion}" else "##{@newSha.substr(0, 8)}"
-        console.error("Updating #{@type} #{@pack.name} to #{version} failed", error.stack ? error, error.stderr)
+        atom.assert false, "Package update failed", (assertionError) =>
+          assertionError.metadata = {
+            type: @type,
+            name: pack.name,
+            version: version,
+            errorMessage: error.message,
+            errorStack: error.stack,
+            errorStderr: error.stderr
+          }
+        console.error("Updating #{@type} #{pack.name} to #{version} failed:\n", error, error.stderr ? '')
 
   uninstall: ->
     @packageManager.uninstall @pack, (error) =>

